projects/container_bench/testing/exec_time.py

```diff
@@ -69,33 +275,47 @@ def execute_command(command_list, stop_event, monitor_thread):
                execution_time (float): Time taken to execute the command in seconds.
     """
     monitor_thread.start()
+    read_power_event.set()
     start_time = time.perf_counter()
-    with subprocess.Popen(
-        command_list,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        text=True,
-        bufsize=1,
-        universal_newlines=True
-    ) as process:
-        stdout, stderr = process.communicate()
-        process.wait()
-    end_time = time.perf_counter()
-    stop_event.set()
-    monitor_thread.join()
+    try:
+        with subprocess.Popen(
+            command_list,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            bufsize=1,
+        ) as process:
+            stdout, stderr = process.communicate()
+            process.wait()
+            return_code = process.returncode
+        end_time = time.perf_counter()
+    except Exception as e:
+        stdout, stderr, return_code, end_time = "", str(e), -1, time.perf_counter()
+    finally:
+        stop_event.set()
+        read_power_event.clear()
+        monitor_thread.join()
 
     execution_time = end_time - start_time
-    return_code = process.returncode
 
     return stdout, stderr, return_code, execution_time
 
 
 def write_to_file(filepath, content):
-    with open(filepath, 'a') as f:
+    with open(filepath, 'a', encoding='utf-8') as f:
```

Comment:
not important, but I don't think you need that, with Python3 UTF8 is the default AFAIR